### PR TITLE
Label tests with openshift crafted API groups

### PR DESCRIPTION
**Pull request by ingvagabund**
Revering https://github.com/openshift/origin/pull/27378

Waiting for https://github.com/openshift/release/pull/31818

In addition to original https://github.com/openshift/origin/pull/27368 `test/extended/util/annotate/rules.go` have been updated to take into account the new labels as well

jira: https://issues.redhat.com/browse/WRKLDS-488
jira: https://issues.redhat.com/browse/OCPBUGS-600

### CODE REVIEW

File: test/extended/scheduling/pods.go [hunk 1]
```diff
@@ -89,33 +87,6 @@ func (p requirePodsOnDifferentNodesTest) getDeploymentPods(oc *exutil.CLI) []*co
 }
 
 func (p requirePodsOnDifferentNodesTest) run(oc *exutil.CLI) {
-	ctx := context.TODO()
```
Comment (pacevedom) [inline test/extended/scheduling/pods.go:92]: Do we have upgrade tests where this might be running?

**Comment by deads2k** [comment]
/approve

@ingvagabund get up with TRT to figure out what they want to see pass on this PR before merge.

File: test/extended/cli/env.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ var _ = g.Describe("[sig-cli] oc env", func() {
 		oc              = exutil.NewCLI("oc-env")
 	)
 
-	g.It("can set environment variables", func() {
+	g.It("can set environment variables [apigroup:apps.openshift.io][apigroup:build.openshift.io]", func() {
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply test/extended/cli/env.go:19]: Right, ImageStream in test-buildcli.json

File: test/extended/scheduling/pods.go (hunk 1 above)
Comment (ingvagabund) [reply test/extended/scheduling/pods.go:92]: Only two jobs I could find:
- ./release/openshift-release-master__nightly-4.12-upgrade-from-stable-4.10.yaml
- ./release/openshift-release-master__ci-4.12-upgrade-from-stable-4.11-from-stable-4.10.yaml

Checking the latter openshift-tests built from 4.12/master is run inside the job.

Comment (ingvagabund) [reply test/extended/scheduling/pods.go:92]: https://github.com/openshift/origin/pull/26899 was merged in 4.11/master. Meaning 4.10 should comply with the test as well.

File: test/extended/cli/help.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ var _ = g.Describe("[sig-cli] oc help", func() {
 
 	oc := exutil.NewCLIWithoutNamespace("oc-help")
 
-	g.It("works as expected", func() {
+	g.It("works as expected [apigroup:user.openshift.io][apigroup:project.openshift.io]", func() {
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply test/extended/cli/help.go]: Removed

File: test/extended/cli/env.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ingvagabund) [reply test/extended/cli/env.go]: Added

**Comment by stbenjam** [comment]
/hold

Let's make sure e2e-metal-ipi-ovn-ipv6 passes though, which would start failing if we got things wrong.

**Comment by stbenjam** [comment]
https://github.com/openshift/origin/blob/c3174c86daba77e408662bb74e481321f411a0dd/test/extended/util/annotate/rules.go#L144 needs updating, too

**Comment by stbenjam** [comment]
Looks correct now

/hold cancel
/lgtm

**Comment by ingvagabund** [comment]
Blocked by https://github.com/openshift/origin/pull/27394

**Comment by deads2k** [comment]
/skip ci/prow/e2e-gcp-builds

the build job seems to be failing on many PRs on the same test.

**Comment by deads2k** [comment]
/override ci/prow/e2e-aws-ovn-fips
/override ci/prow/e2e-gcp-ovn-upgrade 

fips failed on pathological events (not this PR) and the ugprade failed on disruption (not this PR).

**Comment by deads2k** [comment]
/override ci/prow/e2e-aws-image-registry

previously worked

**Comment by stbenjam** [comment]
/retitle OCPBUGS-600: Label tests with openshift crafted API groups

Making jira make the link.

**Comment by stbenjam** [comment]
Looks like aws-proxy job is failing since it has a partial match in the release repo skips we didn't find in earlier grepping.

https://github.com/openshift/release/pull/32243 may fix it.

**Pull request merged**